File: frontend/src/app/fascicolo/creazione-fascicolo/dto/ConduzioneDto.ts
export class ConduzioneDto {
    superficieCondotta: string;
    codiceTipoConduzione: string;
    descrizioneTipoConduzione: string;
    dataInizioConduzione: string;
    dataFineConduzione: string;
    codiceFiscaleProprietarioList: string[];
    datiParticella: ParticellaDto;
    caratteristicheZona: CaratteristicheZonaDto;
    documentiConduzione: DocumentiConduzioneDto[];

    static toDtos(incs: any): ConduzioneDto[] {
        if (!incs) {
            return null;
        }
        const list: ConduzioneDto[] = [];
        for (const inc of incs) {
            if (!inc) {
                continue;
            }
            list.push(ConduzioneDto.toDto(inc));
        }
        return list;
    }

    static toDto(inc: any): ConduzioneDto {
        if (!inc) {
            return null;
        }
        const dati: ConduzioneDto = new ConduzioneDto();
        dati.superficieCondotta = inc.superficieCondotta;
        dati.codiceTipoConduzione = inc.codiceTipoConduzione;
        dati.descrizioneTipoConduzione = inc.descrizioneTipoConduzione;
        dati.dataInizioConduzione = inc.dataInizioConduzione;
        dati.dataFineConduzione = inc.dataFineConduzione;
        dati.codiceFiscaleProprietarioList = inc.codiceFiscaleProprietarioList;
        dati.datiParticella = ParticellaDto.toDto(inc.datiParticella);
        dati.caratteristicheZona = CaratteristicheZonaDto.toDto(inc.caratteristicheZona);
        dati.documentiConduzione = DocumentiConduzioneDto.toDtos(inc.documentiConduzione);
        return dati;
    }
}

export class ParticellaDto {
    provincia: string;
    comune: string;
    sezione: string;
    foglio: string;
    particella: string;
    subalterno: string;

    static toDto(inc: any): ParticellaDto {
        if (!inc) {
            return null;
        }
        const dati: ParticellaDto = new ParticellaDto();
        dati.provincia = inc.provincia;
        dati.comune = inc.comune;
        dati.sezione = inc.sezione;
        dati.foglio = inc.foglio;
        dati.particella = inc.particella;
        dati.subalterno = inc.subalterno;
        return dati;
    }
}

export class CaratteristicheZonaDto {
    casiParticolari: string;
    casiParticolariDescrizione: string;
    flagGiust: string;
    flagGiustDescrizione: string;
    codiZVN: string;
    codiZVNDescrizione: string;

    static toDto(inc: any): CaratteristicheZonaDto {
        if (!inc) {
            return null;
        }
        const dati: CaratteristicheZonaDto = new CaratteristicheZonaDto();
        dati.casiParticolari = inc.casiParticolari;
        dati.casiParticolariDescrizione = inc.casiParticolariDescrizione;
        dati.flagGiust = inc.flagGiust;
        dati.flagGiustDescrizione = inc.flagGiustDescrizione;
        dati.codiZVN = inc.codiZVN;
        dati.codiZVNDescrizione = inc.codiZVNDescrizione;
        return dati;
    }
}

export class DocumentiConduzioneDto {
    dataRilascio: string;
    dataScadenza: string;
    numeroDocumento: string;
    tipoDocumento: string;

    static toDtos(incs: any): DocumentiConduzioneDto[] {
        if (!incs) {
            return null;
        }
        const list: DocumentiConduzioneDto[] = [];
        for (const inc of incs) {
            if (!inc) {
                continue;
            }
            list.push(DocumentiConduzioneDto.toDto(inc));
        }
        return list;
    }

    static toDto(inc: any): DocumentiConduzioneDto {
        if (!inc) {
            return null;
        }
        const dati: DocumentiConduzioneDto = new DocumentiConduzioneDto();
        dati.dataRilascio = inc.dataRilascio;
        dati.dataScadenza = inc.dataScadenza;
        dati.numeroDocumento = inc.numeroDocumento;
        dati.tipoDocumento = inc.tipoDocumento;
        return dati;
    }
}

File: backend/a4gproxy/a4gproxy-integration/src/main/java/it/tndigitale/a4g/proxy/dto/sian/conduzioneterreno/ConduzioneDto.java
package it.tndigitale.a4g.proxy.dto.sian.conduzioneterreno;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

import java.io.Serializable;
import java.util.List;

@ApiModel(description = "DTO della conduzione del terreno recuperato dal SIAN")
public class ConduzioneDto implements Serializable{

	@ApiModelProperty(value = "SuperficieCondotta")
	private String superficieCondotta;
	@ApiModelProperty(value = "CodiceTipoConduzione")
	private String codiceTipoConduzione;
	@ApiModelProperty(value = "descrizioneTipoConduzione")
	private String descrizioneTipoConduzione;
	@ApiModelProperty(value = "DataInizioConduzione")
	private String dataInizioConduzione;
	@ApiModelProperty(value = "DataFineConduzione")
	private String dataFineConduzione;
	@ApiModelProperty(value = "Elenco Codice Fiscale dei Proprietari")
	private List<String> codiceFiscaleProprietarioList;
	@ApiModelProperty(value = "Dati Particella")
	private ParticellaDto datiParticella;
	@ApiModelProperty(value = "Caratteristiche della zona")
	private CaratteristicheZonaDto caratteristicheZona;
	@ApiModelProperty(value = "Elenco documenti giustificativi sulla conduzione terreni")
	private List<DocumentoConduzioneDto> documentiConduzione;

	public String getSuperficieCondotta() {
		return superficieCondotta;
	}

	public void setSuperficieCondotta(String superficieCondotta) {
		this.superficieCondotta = superficieCondotta;
	}

	public String getCodiceTipoConduzione() {
		return codiceTipoConduzione;
	}

	public void setCodiceTipoConduzione(String codiceTipoConduzione) {
		this.codiceTipoConduzione = codiceTipoConduzione;
	}

	public String getDescrizioneTipoConduzione() {
		return descrizioneTipoConduzione;
	}

	public void setDescrizioneTipoConduzione(String descrizioneTipoConduzione) {
		this.descrizioneTipoConduzione = descrizioneTipoConduzione;
	}

	public String getDataInizioConduzione() {
		return dataInizioConduzione;
	}

	public void setDataInizioConduzione(String dataInizioConduzione) {
		this.dataInizioConduzione = dataInizioConduzione;
	}

	public String getDataFineConduzione() {
		return dataFineConduzione;
	}

	public void setDataFineConduzione(String dataFineConduzione) {
		this.dataFineConduzione = dataFineConduzione;
	}

	public List<String> getCodiceFiscaleProprietarioList() {
		return codiceFiscaleProprietarioList;
	}

	public void setCodiceFiscaleProprietarioList(List<String> codiceFiscaleProprietarioList) {
		this.codiceFiscaleProprietarioList = codiceFiscaleProprietarioList;
	}

	public ParticellaDto getDatiParticella() {
		return datiParticella;
	}

	public void setDatiParticella(ParticellaDto datiParticella) {
		this.datiParticella = datiParticella;
	}

	public CaratteristicheZonaDto getCaratteristicheZona() {
		return caratteristicheZona;
	}

	public void setCaratteristicheZona(CaratteristicheZonaDto caratteristicheZona) {
		this.caratteristicheZona = caratteristicheZona;
	}

	public List<DocumentoConduzioneDto> getDocumentiConduzione() {
		return documentiConduzione;
	}

	public void setDocumentiConduzione(List<DocumentoConduzioneDto> documentiConduzione) {
		this.documentiConduzione = documentiConduzione;
	}
}
